bootloader/bootloader.c

```diff
@@ -139,11 +139,15 @@ void __attribute__((noreturn)) __stack_chk_fail(void)
 
 int main(void)
 {
-	__stack_chk_guard = random32();
 	setup();
 	memory_protect();
 	oledInit();
 
+	// to be extra careful and heed the STM32F205xx Reference manual, Section 20.3.1
+	// we don't use the first random number generated after setting the RNGEN bit in setup
+	random32();
+	__stack_chk_guard = random32(); // this supports compiler provided unpredictable stack protection checks
```

Comment:
nah, can only modify the canary within main because it doesn't return. there's literally nowhere to even return to since ```reset_handler``` doesn't push anything onto the stack, it just branches. going into main, the MSP points just after the end of SRAM. ```load_app``` does the same thing when going from bootloader to firmware.

i verified that with my latest commit, and then adding the following patch, the stack smashing message displays:
```
diff --git a/setup.c b/setup.c
index 2537a01..e73fe5c 100644
--- a/setup.c
+++ b/setup.c
@@ -23,6 +23,8 @@
 #include <libopencm3/stm32/f2/rng.h>
 #include "rng.h"
 
+extern uint32_t __stack_chk_guard;
+
 void setup(void)
 {
        // setup clock
@@ -79,6 +81,7 @@ void setupApp(void)
        // "Each subsequent generated random number has to be compared with the previously generated
        // number. The test fails if any two compared numbers are equal (continuous random number generator test)."
        random32();
+       __stack_chk_guard = 1;
        // hotfix for old bootloader
        gpio_mode_setup(GPIOA, GPIO_MODE_INPUT, GPIO_PUPD_NONE, GPIO9);
        spi_init_master(SPI1, SPI_CR1_BAUDRATE_FPCLK_DIV_8, SPI_CR1_CPOL_CLK_TO_0_WHEN_IDLE, SPI_CR1_CPHA_CLK_TRANSITION_1, SPI_CR1_DFF_8BIT, SPI_CR1_MSBFIRST);
```